### PR TITLE
envoy: cleanup istio specifics

### PR DESCRIPTION
**Pull request by mhofstetter**
As a follow up of #30130, this commit removes Istio sidecar specific implementations and/or comments from the Envoy module.

### CODE REVIEW

**Review by sayboras — APPROVED** [review]
Thanks a lot :heavy_check_mark:

File: pkg/envoy/embedded_envoy.go [hunk 1]
```diff
@@ -104,7 +104,6 @@ func startEmbeddedEnvoy(config embeddedEnvoyConfig) (*EmbeddedEnvoy, error) {
 	bootstrapFilePath := filepath.Join(config.runDir, "envoy", "bootstrap.pb")
 
 	writeBootstrapConfigFile(bootstrapConfig{
-		// Use the same structure as Istio's pilot-agent for the node ID: nodeType~ipAddress~proxyId~domain
```
Comment (jrajahalme) [inline pkg/envoy/embedded_envoy.go:107]: I would retain a comment about the `nodeId` format (inherited from Istio), or then simplify it to something like "Cilium-host-proxy" etc.

File: pkg/envoy/xds/server.go [hunk 2]
```diff
@@ -276,7 +277,7 @@ func (s *Server) processRequestStream(ctx context.Context, streamLog *logrus.Ent
 				id := req.GetNode().GetId()
 				streamLog = streamLog.WithField(logfields.XDSClientNode, id)
 				var err error
-				nodeIP, err = IstioNodeToIP(id)
+				nodeIP, err = EnvoyNodeToIP(id)
```
Comment (jrajahalme) [inline pkg/envoy/xds/server.go]: Maybe trace down where this `nodeIP` is used and why?

**Review by jrajahalme — CHANGES_REQUESTED** [review]
May want to reformat the `nodeId` we use in xDS.

Also, the endpoint package still has related code/comments that would benefit from a clean-up:
- updateNetworkPolicy()
- WaitForFirstRegeneration()
- HasSidecarProxy()

proxy package mock endpoint also has code related to "SidecarProxy"

File: pkg/envoy/xds/server.go (hunk 2 above)
Comment (mhofstetter) [reply pkg/envoy/xds/server.go]: I did. It's used in the `AckingResourceMutatorWrapper` to be able to distinguish between acked versions of different Envoy instances (probably Cilium Node Proxy and Envoy).

- https://github.com/cilium/cilium/blob/main/pkg/envoy/xds/ack.go#L314-L316
- https://github.com/cilium/cilium/blob/main/pkg/envoy/xds/ack.go#L104-L107

Raises the decision whether we want to drop this functionalty or keep (for potential use-cases in the future). if we decide to delete it - i tend to do it in a separate PR if this is OK for you.

(Another case is logging it as field in the xds resource watcher - https://github.com/cilium/cilium/blob/main/pkg/envoy/xds/watcher.go#L90 - nothing that can't be deleted)

File: pkg/envoy/embedded_envoy.go (hunk 1 above)
Comment (mhofstetter) [reply pkg/envoy/embedded_envoy.go:107]: i added back a small comment about having this inherited from Istio. I agree that we could simplify the name. Especially if we decide to delete the need for "extracting the "IP" from the Envoy Node ID.

But i'd like to have this in a separate PR to keep the context smaller (the other would also impact the DaemonSet mode where we set the node id in the Helm YAMLs.

**Comment by mhofstetter** [comment]
> May want to reformat the `nodeId` we use in xDS.
> 
> Also, the endpoint package still has related code/comments that would benefit from a clean-up:
> 
> * updateNetworkPolicy()
> * WaitForFirstRegeneration()
> * HasSidecarProxy()
> 
> proxy package mock endpoint also has code related to "SidecarProxy"

Thanks Jarno! I removed/adjusted these places too.

I even found that some of these Endpoint/XDS ifaces are no longer used. Will open a follow up PR for these.

PTAL.

File: pkg/envoy/xds/server.go (hunk 2 above)
Comment (mhofstetter) [reply pkg/envoy/xds/server.go]: :x: looks like we can't delete this - it looks like proxylib also uses it's own IP

https://github.com/cilium/cilium/pull/9395 (PR that introduced the map)

https://github.com/cilium/proxy/blob/main/proxylib/proxylib/instance.go#L57

**Review by jrajahalme — APPROVED** [review]

File: pkg/envoy/xds/server.go (hunk 2 above)
Comment (jrajahalme) [reply pkg/envoy/xds/server.go]: Right, proxylib has it's own xDS client.

**Pull request merged**